compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -244,44 +244,36 @@ object TypeOps:
 
     def mergeRefinedOrApplied(tp1: Type, tp2: Type): Type = {
       def fail = throw new AssertionError(i"Failure to join alternatives $tp1 and $tp2")
-      def fallback = tp2 match
+
+      tp2 match
```

Comment:
This is now done first instead of last. Why the change?

Reply:
I want to handle the case of `tp1` and `tp2` are both AnyTypes and `eq` to each other first.